sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -35,3 +36,47 @@ case class Size(child: Expression) extends UnaryExpression with ExpectsInputType
     nullSafeCodeGen(ctx, ev, c => s"${ev.primitive} = ($c).size();")
   }
 }
+
+case class ArrayContains(left: Expression, right: Expression) extends BinaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = BooleanType
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, AnyDataType)
```

Comment:
Just be sure we are in the same page, the second dataType should be identical with the inner type of the ArrayType (first dataType). 

`bigint`(LongType) is not acceptable, if the dataType of first argument is ArrayType(IntegerType, _), but NullType will be OK, as the `NullType` is implicit acceptable for all of the expected data types.

It will be nice if you can try that in the unit test.